updates for client_id and tech_acct_id

mbonish opened the pull request:
Updated documentation and config files to reflect changes to UMAPI. 
Updated api_key to client_id and tech_acct to tech_acct_id.

adorton-adobe committed:
version bump

vossen-adobe committed:
Merge remote-tracking branch 'ust/v2' into v2

adorton-adobe committed:
fix example bundle regression

mbonish committed:
updates for client_id and tech_acct_id

updated documentation and config files to reflect changes to UMAPI. Updated api_key to client_id and tech_acct to tech_acct_id

vossen-adobe committed:
Merge branch 'v2' into dx_update

vossen-adobe committed:
don't rename travis

vossen-adobe reviewed (changes requested):
We need to maintain backwards compatability - use tech_acct_id if it exists, but otherwise default to tech_acct.  E.g., something like:

tech_field = 'tech_acct_id' if 'tech_acct_id' in enterprise_config else 'tech_acct'
integration_builder.require_string_value(tech_field)

And same for api key.

In the current state, we cannot run sync with existing umapi files.

mbonish committed:
Updates for backwards compatibility

Made changes in appropriate spots so client_id/ api_key and tech_acct/tech_acct_id work.

vossen-adobe commented on user_sync/connector/directory_adobe_console.py:
```diff
@@ -100,7 +101,9 @@ def __init__(self, caller_options):
         self.org_id = org_id = integration_options['org_id']
-        auth_dict = make_auth_dict(self.name, enterprise_config, org_id, integration_options['tech_acct'], logger)
+
+        tech_field = 'tech_acct_id' if 'tech_acct_id' in integration_options else 'tech_acct'
```
Redundant line?

vossen-adobe commented on user_sync/connector/umapi.py:
```diff
@@ -81,7 +82,8 @@ def __init__(self, name, caller_options):
         ims_host = server_options['ims_host']
         self.org_id = org_id = enterprise_options['org_id']
-        auth_dict = make_auth_dict(self.name, enterprise_config, org_id, enterprise_options['tech_acct'], logger)
+        tech_field = 'tech_acct_id' if 'tech_acct_id' in enterprise_options else 'tech_acct'
```
Redundant line?

vossen-adobe commented on user_sync/connector/directory_adobe_console.py:
```diff
@@ -100,7 +101,9 @@ def __init__(self, caller_options):
         ims_host = server_options['ims_host']
         self.org_id = org_id = integration_options['org_id']
-        auth_dict = make_auth_dict(self.name, enterprise_config, org_id, integration_options['tech_acct'], logger)
+
```
don't add blank line

vossen-adobe commented on user_sync/connector/umapi_util.py:
```diff
@@ -3,10 +3,11 @@
+    api_field = 'client_id' if 'client_id' in config else "api_key"
     auth_dict = {
         'org_id': org_id,
-        'tech_acct_id': tech_acct,
```
This line should not be changed - must be tech_acct_id since this is what umapi client expects.

vossen-adobe reviewed (changes requested):
can you also test with the secure_xx_key format to make sure it still works?

mbonish committed:
clean up

-removed redundant code
-fixed secure_client_id

mbonish committed:
fixed issue with api_field

fixed logical error in ternary

vossen-adobe commented:
Gonna close this since it should be on api now
